### PR TITLE
Adjust swift target based on CMAKE_OSX_ARCHITECTURE

### PR DESCRIPTION
**Pull request by dmbryson**

### CODE REVIEW

File: products/llbuildSwift/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ endif()
 
 # TODO(compnerd) move both of these outside of the CMake into the invocation
 if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
-  add_compile_options(-target x86_64-apple-macosx10.10)
+  add_compile_options(-target ${CMAKE_OSX_ARCHITECTURE}-apple-macosx)
```
Comment (shahmishal) [inline products/llbuildSwift/CMakeLists.txt:20]:
```suggestion
  add_compile_options(-target ${CMAKE_OSX_ARCHITECTURES}-apple-macosx10.10)
```

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
https://github.com/apple/swift/pull/33490

@swift-ci test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
Please hold off from merging this until apple/swift#33490 has been merged.

**Comment by shahmishal** [comment]
@swift-ci please smoke test

**Pull request merged**